### DISCUSSION TITLE
The self-compiled version always prompts that libsciter-gtk.so cannot be found

### DISCUSSION THREAD

**Discussion by kero990** [rustdesk/rustdesk#2348] [General]
Compile rustdesk on a pc with arm64 linux environment, but no matter whether you compile the debug version or the release version, you will be prompted when running
'libsciter-gtk.so' was not found neither in PATH nor near the current executable.
 

>  'libsciter-gtk.so' was not found neither in PATH nor near the current executable.
>   Please verify that Sciter SDK is installed and its binaries (from bin.lnx) are available in PATH.', /root/.cargo/git/checkouts/rust-sciter-06aa50f9c0fcf3d6/82025b9/src/lib.rs:359:21
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

And no matter where I put the libsciter-gtk.so file, /uer/lib or other lib directories, and the current directory of the program, there are the above prompts.
In addition, the path of the last section of the error report is the path when I compiled it. Why use the compiled binary file, and the error is still this path?

**Comment by grummbeer** [comment]
https://rustdesk.com/docs/en/dev/build/linux/#dependencies

Download from https://github.com/c-smile/sciter-sdk/tree/master/bin.lnx and copy it to `target/(release|debug)/libsciter-gtk.so`

**Comment by kero990** [comment]
I'm sure I downloaded libsciter-gtk.so correctly and put it in the target/debug directory, but apparently that doesn't work. I saw in another discussion that you said that sciter-sdk is not easy to use under arm64, I don't know if it is what I encountered.

Not sure if anyone has tested compiling and using it on arm64. In addition, because my real environment is a little special, debian10 on arm64, and some drivers are not open source (that is, bound to the 4.19 kernel), so many programs will not run on my machine because the version of glibc is too low, I have to Compilation is mostly application.

I don't know whether the .so file of sciter-sdk will be affected by the system environment, and whether it can work in the standard environment of debian10. Maybe I have to do some more testing.

Looking forward to your official arm64 release.

**Comment by 7th-fret** [comment]
Take a look at src/ui.rs, lines 59, 60

        let dir = "/usr";
        sciter::set_library(&(prefix + dir + "/lib/rustdesk/libsciter-gtk.so")).ok();

As you can see it seeks libsciter-gtk.so under absolute path /usr/lib/rustdesk/
Try to make a folder named rustdesk under /usr/lib and copy libsciter-gtk.so there.
I had the same problem and this was the solution.
Also, make sure that you download the libsciter-gtk.so for arm64 platform.

**Comment by basinilya** [comment]
Had a similar problem with the official 32bit deb on 64bit os. Turned out, the hidden error was:
```
dlopen failed:
libuuid.so.1: cannot open shared object file: No such file or directory
```

Fixed by running: `sudo apt install libuuid1:armhf`.
<details>
  <summary>troubleshooting process</summary>

  ```
  $ sudo apt install gcc-arm-linux-gnueabihf
  $ arm-linux-gnueabihf-gcc test.c
  $ ./a.out
  ELF Header
  ==========
  Class      : Other
  Machine    : 40
  Entry      : 0x197b18
  PH count   : 58316
  SH count   : 58316
  
  dlopen failed:
  libuuid.so.1: cannot open shared object file: No such file or directory
  ```

  [test.c.txt](https://github.com/user-attachments/files/28699044/test.c.txt)

</details>
